openmetadata-ui/src/main/resources/ui/src/components/CustomEntityDetail/AddCustomProperty/AddCustomProperty.tsx

```diff
@@ -11,155 +11,137 @@
-    const nameRegEx = /^[a-z][a-zA-Z0-9]+$/;
+  const propertyTypeOptions = useMemo(() => {
+    const supportedTypes = propertyTypes.filter((property) =>
+      SUPPORTED_FIELD_TYPES.includes(property.name)
+    );
 
-    return nameRegEx.test(name);
-  };
+    return map(supportedTypes, (type) => ({
+      key: type.name,
+      label: startCase(type.displayName || type.name),
+      value: type.id,
+    }));
+  }, [propertyTypes]);
 
-  const validateType = (type: string) => {
-    return Boolean(type);
+  const fetchPropertyType = async () => {
+    try {
+      const response = await getTypeListByCategory(Category.Field);
+      setPropertyTypes(response.data);
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    }
   };
 
-  const handleError = (flag: boolean, property: string) => {
-    const message =
-      property === 'name'
-        ? t('message.invalid-property-name')
-        : t('message.field-text-is-required', { fieldText: t('label.type') });
-
-    setFormErrorData((preVdata) => ({
-      ...preVdata,
-      [property]: !flag ? message : '',
-    }));
+  const fetchTypeDetail = async (typeFQN: string) => {
+    try {
+      const response = await getTypeByFQN(typeFQN);
+      setTypeDetail(response);
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    }
   };
 
-  const onChangeHandler = (
-    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
-  ) => {
-    const { name, value } = e.target;
-
-    switch (name) {
-      case 'name': {
-        const newData = { ...formData, name: value };
-        const isValidName = validateName(value);
-        handleError(isValidName, 'name');
-        setFormData(newData);
+  const handleCancel = useCallback(() => history.goBack(), [history]);
 
-        break;
-      }
-      case 'type': {
-        const newData = { ...formData, type: value };
-        const isValidType = validateType(value);
-        handleError(isValidType, 'type');
-        setFormData(newData);
+  const handleFieldFocus = useCallback((event: FocusEvent<HTMLFormElement>) => {
+    const isDescription = event.target.classList.contains('ProseMirror');
 
-        break;
-      }
+    setActiveField(isDescription ? 'root/description' : event.target.id);
+  }, []);
 
-      default:
-        break;
+  const handleSubmit = async (data: CustomProperty) => {
+    if (isUndefined(typeDetail)) {
+      return;
     }
-  };
 
-  const onCancel = () => {
-    history.goBack();
-  };
-
-  const onSave = () => {
-    const isValidName = validateName(formData.name);
-    const isValidType = validateType(formData.type);
-    if (isValidName && isValidType) {
-      const propertyData = {
-        description: getDescription(),
-        name: formData.name,
+    try {
+      setIsCreating(true);
+      await addPropertyToEntity(typeDetail?.id ?? '', {
+        ...data,
         propertyType: {
-          id: formData.type,
+          id: data.propertyType as unknown as string,
```

Comment:
why we need to typeCast it as unknown as string?

Reply:
In submit handler we have the data type as `CustomProperty` and in `CustomProperty` the `propertyType` is entity reference, but from the form context we are getting `propertyType` as a string, so that's why we need to typecast it.

Reply:
Update: I have refactored the type and removed the type-casting logic.